main.go

```diff
@@ -45,22 +52,40 @@ func main() {
 }
 
 func (cs *cmdServe) Run() error {
-	withLogger := serve.WithLogger(log.NewLogger(os.Stderr, cs.LogLevel))
-	withProtosets := serve.WithProtosets(cs.ProtoSet...)
+	logger := log.NewLogger(os.Stderr, cs.LogLevel)
+	opts, err := cs.getServerOptions(logger)
+	if err != nil {
+		return err
+	}
 	dirs := serve.NewFSFromDirs(cs.Dirs...)
-	s, err := serve.NewServer(serve.JsonnetEvaluator(), dirs, withLogger, withProtosets)
+	s, err := serve.NewServer(serve.JsonnetEvaluator(), dirs, opts...)
 	if err != nil {
 		return err
 	}
 
 	if cs.HTTP {
-		h := httprule.NewServer(s.Files, s.UnknownHandler)
+		h := httprule.NewServer(s.Files, s.UnknownHandler, logger)
 		s.SetHTTPHandler(h)
 	}
 
 	return s.ListenAndServe(cs.Listen)
 }
 
+func (cs *cmdServe) getServerOptions(logger log.Logger) ([]serve.Option, error) {
+	var opts []serve.Option
```

Comment:
nit: This could be:
```
    opts := []serve.Option{serve.Withlogger(logger), serve.WithProtosets(cs.ProtoSet...)}
```

i.e. no need to append to a slice unconditionally when you have just created the slice.